platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/output/internal/TransformingOutputWire.java

```diff
@@ -43,27 +45,32 @@ public class TransformingOutputWire<IN, OUT> extends ForwardingOutputWire<IN, OU
     private final List<Consumer<OUT>> forwardingDestinations = new ArrayList<>();
 
     private final Function<IN, OUT> transform;
-    private final Consumer<IN> cleanup;
+    private final Consumer<IN> inputCleanup;
+    private final Consumer<OUT> outputCleanup;
 
     /**
      * Constructor.
      *
-     * @param model       the wiring model containing this output wire
-     * @param name        the name of the output wire
-     * @param transformer the function to transform the data from the input type to the output type. Is called once per
-     *                    output per data item. If this method returns null then the data is not forwarded.
-     * @param cleanup     an optional method that is called after the data is forwarded to all destinations. The
-     *                    original data is passed to this method. Ignored if null.
+     * @param model         the wiring model containing this output wire
+     * @param name          the name of the output wire
+     * @param transformer   the function to transform the data from the input type to the output type. Is called once
+     *                      per output per data item. If this method returns null then the data is not forwarded.
+     * @param inputCleanup  an optional method that is called on input data after the data is forwarded to all
+     *                      destinations. The original data is passed to this method. Ignored if null.
+     * @param outputCleanup an optional method that is called on output data if it is rejected by a destination. This is
+     *                      possible if offer soldering is used and the destination declines to take the data.
      */
     public TransformingOutputWire(
```

Comment:
I feel like this method should be removed. We can still support the lambda approach with a helper class. @cody-littley wdyt?

Reply:
Agreed in zoom call

Reply:
done